### ISSUE TITLE
Add Standardizer preprocessor to netcdf dataset enhancements

### ISSUE THREAD

**Issue by cssjessica** [Unidata/netcdf-java#1206]
[Standardization](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html#sklearn.preprocessing.StandardScaler.transform) is a common preprocessing step for AI/ML inputs. So, the goal is to have a raw and a preprocessed version of datasets in the TDS Catalog. The Standardizer preprocessing can be achieved through NcML as

```
<netcdf xmlns="http://www.unidata.ucar.edu/namespaces/netcdf/ncml-2.2" enhance="all">
  <variable name="scaledVar">
    <attribute name="standardize"/>
  </variable>
</netcdf>
```

- [x] Compute the mean and standard deviation for scaling
- [x] Perform scaling using Scale Offset
- [x] Add to set of enhancements and apply when correct attribute exists
- [x] Test with a dataset on thredds-test server
- [x] Add notebooks for visualization
- [x] Test performance

**Comment by haileyajohnson** [comment]
This is great!
I'd like to talk about the name of the enhancement before it's too far along. To me "scaler" is a bit confusing because it's so similar to "scalar". So maybe just `Standardize` or `Normalize`?

**Comment by cssjessica** [comment]
Yes! @tdrwenski also commented about it being a bit confusing. I'll change to `Normalize` then.

**Comment by tdrwenski** [comment]
I think "normalize" makes it sound like it is rescaled to unit length so I think I prefer "standardize". So would we call the class that performs the operation a `Standardizer` (since its good to use a noun for the class name)?

**Comment by cssjessica** [comment]
That's true! With standardization, the values don't have a bounding range

**Comment by haileyajohnson** [comment]
awesome, thanks @cssjessica !
